### DISCUSSION TITLE
Is it still recommended to restart scheduler periodically?

### DISCUSSION THREAD

**Discussion by geobio** [apache/airflow#19219] [Q&A] [answered]
Years ago in earlier versions of airflow, it was a known issue that a long-running airflow scheduler could run into issues and not function well.

Therefore, it seemed to be common practice and recommended to restart the scheduler periodically. For example: https://medium.com/handy-tech/airflow-tips-tricks-and-pitfalls-9ba53fba14eb#.744i823i6

Quote:
```
The scheduler should be restarted frequently
In our experience, a long running scheduler process, at least with the CeleryExecutor, ends up not scheduling some tasks. We still don’t know the exact cause, unfortunately.
Fortunately, airflow has a built-in workaround in the form of the — num_runs flag. It specifies a number of iterations for the scheduler to run of its loop before it quits. We’re running it with 10 iterations, Airbnb runs it with 5. Note that this will cause problems when using the LocalExecutor.
```

I would think that since airflow was initially released with these issues, the schedule can now remaining running indefinitely without issue. However I didn't find any concrete answers on the internet regarding this. Is it still recommended to restart schedule periodically?

Thank you

**Comment by potiuk** [comment, marked as answer]
No
